src/v2/view-builder/views/webauthn/RequiredFactorWebauthnView.js

```diff
@@ -0,0 +1,139 @@
+import { loc, _, createButton } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import BaseFooter from '../../internals/BaseFooter';
+import BaseFactorView from '../shared/BaseFactorView';
+import CryptoUtil from '../../../../util/CryptoUtil';
+import webauthn from '../../../../util/webauthn';
+
+const Body = BaseForm.extend({
+
+  title: loc('factor.webauthn.biometric', 'login'),
+
+  getUISchema () {
+    const schema = [];
+    // Returning custom array so no fields are displayed for webauthn
+    if (webauthn.isNewApiAvailable()) {
+      const retryButton = createButton({
+        className: 'retry-webauthn button-primary default-custom-button',
+        title: loc('retry', 'login'),
+        click: () => {
+          this.saveForm();
+        }
+      });
+
+      schema.push({
+        View:
+            '<div class="webauthn-verify-text idx-webauthn-verify-text">\
+              <p>{{i18n code="verify.webauthn.biometric.instructions" bundle="login"}}</p>\
+              <div data-se="webauthn-waiting" class="okta-waiting-spinner"></div>\
+            </div>'
+      }, {
+        View: retryButton,
+      });
+    }
+    return schema;
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    if (this.webauthnAbortController) {
+      this.webauthnAbortController.abort();
+      this.webauthnAbortController = null;
+    }
+  },
+
+  noButtonBar: function () {
+    return true;
+  },
+
+  modelEvents: {
+    'request': '_startEnrollment',
+    'error': '_stopEnrollment'
+  },
+
+  saveForm () {
+    this.clearErrors();
+    this.model.trigger('request');
+    this.webauthnAbortController = new AbortController();
+    const factor = this.options.appState.get('factor');
+    const allowCredentials = [{
+      type: 'public-key',
+      id: CryptoUtil.strToBin(factor.contextualData.profile.credentialId)
+    }];
+    const options = {
+      allowCredentials,
+      challenge: CryptoUtil.strToBin(factor.contextualData.challengeData.challenge)
+    };
+    navigator.credentials.get({
+      publicKey: options,
+      signal: this.webauthnAbortController.signal
+    }).then((assertion) => {
+      this.model.set({
+        credentials : {
+          clientData: CryptoUtil.binToStr(assertion.response.clientDataJSON),
+          authenticatorData: CryptoUtil.binToStr(assertion.response.authenticatorData),
+          signatureData: CryptoUtil.binToStr(assertion.response.signature),
+        }
+      });
+      BaseForm.prototype.saveForm.apply(this, [this.model]);
+    }, (error) => {
+      // Do not display if it is abort error triggered by code when switching.
```

Comment:
ah, that's interesting and double confirm `credentials.get` will be rejected when invoke `this.webauthnAbortController.abort();`?